### PR TITLE
Explicit preview

### PR DESCRIPTION
**Pull request by ellenmobify**
Status: **Ready for Review** 
Owner: 
Reviewers: @johnboxall  @jasecode

## Changes
- `browser.preview(url, bundle, callback)`

## Todos:
- [x] Passed linting check (run `grunt lint`)
- [x] Updated README
- [x] Updated CHANGELOG

### Feedback:
_none so far_

## How To Test
- In your test project's package.json, 
`"nightwatch-commands": "mobify/nightwatch-commands#explicit-preview",`
- Change `.preview()` to `.preview('https://www.merlinspotions.com', 'https://localhost:8443/loader.js')` or similar

### CODE REVIEW

File: commands/preview.js [hunk 1]
```diff
@@ -1,106 +1,48 @@
...
 
-    try {
-        var siteConfig = require(path.join(path.resolve('./'), '/system/site.js'));
-    } catch (e) {
-        if (e instanceof Error && e.code === 'MODULE_NOT_FOUND') {
-            console.log('Not using optional /system/site.js.');
-        }
+    if (typeof bundle === 'function') {
+        callback = bundle;
+        bundle = '';
```
Comment (johnboxall) [inline commands/preview.js]: Non-blocking, but in this case I'd usually set `bundle` to something like `null` or `undefined` to make the intention more clear (that it wasn't provided)

File: README.md [hunk 2]
```diff
@@ -180,74 +180,19 @@ this.demoTest = function (browser) {
...
-```
-
-If the project does not have a `site.json` or `site.js` file, this command is equivalent to the `url` protocol command.
+The `preview` command uses http://preview.mobify.com to open a website specified by `url` to preview a given bundle specified by `bundle`. 
 
 Parameter Name | Parameter Type | Description
 -------------  | -------------- | -----------
-url            | String         | _optional_ The URL to preview.
+url            | String         | The URL to preview, equivalent to the Site URL field. 
+bundle         | String         | The bundle URL, equivalent to the Bundle Location field.
```
Comment (johnboxall) [inline README.md]: It's probably worth it to mention this value isn't required, and if it isn't passed a default is used... (as the default also changed)

File: commands/preview.js [hunk 3]
```diff
@@ -1,106 +1,48 @@
...
-
-        return browser.url('https://preview.mobify.com?' + params)
-            .waitForElementPresent('#authorize', 10000, function() {
-                this.click('#authorize', function() {
-                    browser.waitUntilMobified(10000, function(result) {
-                        if (typeof callback === 'function') {
-                            callback.call(browser, result);
-                        }
-                    });
+    return browser.url('https://preview.mobify.com?' + params)
```
Comment (johnboxall) [inline commands/preview.js]: Non-blocking, but would this block be more clear in the chain style? eg.

```js
browser.url(PREVIEW_URL)
    .waitForElementPresent(AUTH_SELECTOR, DEFAULT_TIMEOUT)
    .click(AUTH_SELECTOR, DEFAULT_TIMEOUT)
    .waitUntilMobified(DEFAULT_TIMEOUT)
    .perform(callback.bind(browser) || function(){})
```

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nightwatch-commands",
-  "version": "2.1.0",
+  "version": "3.0.0",
```
Comment (johnboxall) [inline package.json]: 💥

File: commands/preview.js [hunk 5]
```diff
@@ -1,106 +1,48 @@
...
-            url = site.siteUrl;
-        }
+    var params = qs.stringify({'url': url, 'site_folder': bundleUrl});
 
-        // First checks for the URL, otherwise uses the site.siteURL, then makes sure
-        // that there is an http prefix. The preview function doesn't need this, but
-        // the browser.get() method does.
-        url = url || site.siteUrl;
-
-        if (!url.match(/^http/)) {
```
Comment (johnboxall) [inline commands/preview.js]: Just checking – we lost this URL validation in the new code – are we okay with that?

Comment (ellenmobify) [inline commands/preview.js]: This is fine. `url` gets passed as a param to preview.mobify.com, which doesn't require the URL to begin with `HTTP`.

**Pull request merged**